### PR TITLE
Fixed pagination.feautures

### PR DESCRIPTION
**Pull request by AndreaOlivieri**
I solved all problem with pagination. After that I refactor the code

### CODE REVIEW

File: app/assets/javascripts/views/searches/preview.js [hunk 1]
```diff
@@ -6,7 +6,7 @@
   this.Terraling.Searches = this.Terraling.Searches || {};
 
   var searches = this.Terraling.Searches;
-  
+
```
Comment (dej611) [inline app/assets/javascripts/views/searches/preview.js:9]: Turn off your IDE auto-spacing settings, or any PR will have pages and pages with this kind of silly changes.

File: app/assets/javascripts/views/searches/preview.js [hunk 2]
```diff
@@ -216,8 +216,9 @@
       }
     }
 
-    table = tableBuilder.createTable(offset);
-    paginationData = tableBuilder.createPagination(offset);
+    var rows_per_page = json["rows_per_page"] || 25;
```
Comment (dej611) [inline app/assets/javascripts/views/searches/preview.js:219]: When the property can be accessed via string in JS, it is usually better than using the squared brackets - I'll send a PR with an `.eslint` for this in the future.

``` js
var rows_per_page = json.rows_per_page || 25;
```

File: app/assets/javascripts/views/searches/preview.js [hunk 3]
```diff
@@ -245,14 +246,47 @@
...
+
       if(navbarOn){
         enableNavbar(json.type);
       }
-      
+
     }, 500);
   }
 
+  function setDataResultGroups(json){
```
Comment (dej611) [inline app/assets/javascripts/views/searches/preview.js:257]: This and the next function are not used anywhere in the code. It is not clear to me why they are here.

**Review by dej611 — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/views/searches/preview.js (hunk 1 above)
Comment (AndreaOlivieri) [reply app/assets/javascripts/views/searches/preview.js:9]: ok

File: app/assets/javascripts/views/searches/preview.js (hunk 2 above)
Comment (AndreaOlivieri) [reply app/assets/javascripts/views/searches/preview.js:219]: ok

File: app/assets/javascripts/views/searches/preview.js (hunk 3 above)
Comment (AndreaOlivieri) [reply app/assets/javascripts/views/searches/preview.js:257]: because in a PR before I created it, I make some mistake with git, sorry.

**Pull request closed without merging**